aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NonManagedEventLoopGroup.java

```diff
@@ -15,164 +15,22 @@
 
 package software.amazon.awssdk.http.nio.netty.internal;
 
-import io.netty.channel.Channel;
-import io.netty.channel.ChannelFuture;
-import io.netty.channel.ChannelPromise;
-import io.netty.channel.EventLoop;
 import io.netty.channel.EventLoopGroup;
-import io.netty.util.concurrent.EventExecutor;
 import io.netty.util.concurrent.Future;
-import io.netty.util.concurrent.ScheduledFuture;
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 
 /**
  * Decorator around {@link EventLoopGroup} that prevents it from being shutdown. Used when the customer passes in a
  * custom {@link EventLoopGroup} that may be shared and thus is not managed by the SDK.
  */
-public final class NonManagedEventLoopGroup implements EventLoopGroup {
-
-    private final EventLoopGroup delegate;
+public final class NonManagedEventLoopGroup extends DelegatingEventLoopGroup {
 
     public NonManagedEventLoopGroup(EventLoopGroup delegate) {
-        this.delegate = delegate;
-    }
-
-    @Override
-    public boolean isShuttingDown() {
-        return delegate.isShuttingDown();
+        super(delegate);
     }
 
     @Override
     public Future<?> shutdownGracefully() {
         return null;
     }
 
-    @Override
-    public Future<?> shutdownGracefully(long quietPeriod, long timeout, TimeUnit unit) {
-        return null;
-    }
-
-    @Override
-    public Future<Void> terminationFuture() {
-        return null;
-    }
-
-    @Override
-    public void shutdown() {
-    }
-
-    @Override
-    public List<Runnable> shutdownNow() {
```

Comment:
Had that though too. For nonmanaged it's easy but for the reference counting implementation not as much. Thoughts?